src/components/Table/hooks/useTableSort.js

```diff
@@ -1,8 +1,10 @@
+import { camelToSnakeCase } from "neetocist";
 import { mergeLeft } from "ramda";
 import { useHistory } from "react-router-dom";
 
+import { getQueryParams, buildUrl } from "utils";
+
 import { URL_SORT_ORDERS } from "../constants";
-import { camelToSnakeCase, getQueryParams, buildUrl } from "../utils";
 
 const useTableSort = () => {
   const queryParams = getQueryParams();
```

Comment:
I had asked you to go through all code connected with pagination/query params in https://github.com/bigbinary/neeto-ui/pull/1964#discussion_r1387462839
Why is `getQueryParams` still used here?

Reply:
@Amaljith-K _a I noticed this code while self-reviewing and kept it as it is because we are calling `useQueryParams` in `Table/index.jsx`, so whenever the URL query params change, the component will re-render, which in turn will call this hook. That's why I initially kept the code using `getQueryParams` as it is. Please correct me if I'm wrong. Also, I have updated the code now to use `useQueryParams`.

Reply:
It is true logically. It will work.
But it is always best to make everything independent. Making this hook logically dependent on the component not only affects the reusability but also makes the code bug-prone.
For example, if someone removes `useQueryParams` from the component in the future, this hook will no longer function as expected. It is a side effect they won't anticipate.